### PR TITLE
VideoPress: store poster panel data in a local state

### LINKED ISSUE #29846: VideoPress: Limit loop duration instead of starting point of hover effect
As implemented right now, when a loop's duration is too long, the starting point is moved back to compensate:

https://user-images.githubusercontent.com/8486249/229206260-63a3f5e8-3e1f-406b-9d2e-6c67cbd3ab4c.mp4

We should invert that logic to keep the starting point fixed, limiting the loop duration instead.

### LINKED ISSUE #29847: VideoPress: Store hover effect data in local state
The hover effect data is stored directly in the block attributes, but that is causing an interferenc with the poster selector, that resets changes to the hover data when it changes:

https://user-images.githubusercontent.com/8486249/229207403-dd100442-3bcc-4006-93de-b15f539ee7db.mp4

https://user-images.githubusercontent.com/77539/229121922-7af568df-221b-43e5-a405-7efc4be8191b.mp4

### PR DESCRIPTION
**Pull request by retrofox**
This PR stores the poster panel data locally (local state) and propagates them to the dataPoster block attribute.

It fixes an issue where the poster data are not preserved while the user changes them via the UI:

https://user-images.githubusercontent.com/77539/229121922-7af568df-221b-43e5-a405-7efc4be8191b.mp4

Fixes #29846
Fixes #29847

## Proposed changes:

* VideoPress: store poster panel data in a local state

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Go to the block editor
* Create/edit a video block
* Open the Poster panel
* Confirm that all values are preserved when changing by using the UI

https://user-images.githubusercontent.com/77539/229122148-c781e3c7-315c-441e-9f32-15c184b63c8e.mov

### CODE REVIEW

File: projects/packages/videopress/src/client/block-editor/blocks/video/components/poster-panel/index.tsx [hunk 1]
```diff
@@ -94,22 +94,19 @@ export function PosterDropdown( {
...
+
+			// Extend the posterData object to include the media library id and url.
+			posterData: {
+				...attributes.posterData,
+				type: 'media-library',
+				id: image.id,
+				url: image.url,
+			},
+		} );
+	}, [] );
```
Comment (dhasilva) [inline projects/packages/videopress/src/client/block-editor/blocks/video/components/poster-panel/index.tsx]: Shouldn't `attributes` be listed?

Comment (retrofox) [reply projects/packages/videopress/src/client/block-editor/blocks/video/components/poster-panel/index.tsx]: yes, maybe `attributes.posterData`

File: projects/packages/videopress/src/client/block-editor/blocks/video/components/poster-panel/index.tsx [hunk 2]
```diff
@@ -31,9 +37,10 @@ import './style.scss';
  * Types
  */
 import type { AdminAjaxQueryAttachmentsResponseItemProps } from '../../../../../types';
-import type { PosterPanelProps, VideoControlProps, VideoGUID } from '../../types';
+import type { PosterDataProps, PosterPanelProps, VideoControlProps, VideoGUID } from '../../types';
 import type React from 'react';
 
+const MIN_LOOP_DURATION = 3 * 1000;
```
Comment (dhasilva) [inline projects/packages/videopress/src/client/block-editor/blocks/video/components/poster-panel/index.tsx:43]: :brain:

File: projects/packages/videopress/src/client/block-editor/blocks/video/components/poster-panel/index.tsx [hunk 3]
```diff
@@ -318,8 +322,18 @@ function VideoHoverPreviewControl( {
 	onPreviewAtTimeChange,
 	onLoopDurationChange,
 }: VideoHoverPreviewControlProps ): React.ReactElement {
-	const maxLoopDuration = Math.min( MAX_LOOP_DURATION, videoDuration );
-	const maxStartingPoint = videoDuration - loopDuration;
+	const maxStartingPoint = Math.min( videoDuration, videoDuration - MIN_LOOP_DURATION );
+	const maxLoopDuration = Math.min( MAX_LOOP_DURATION, videoDuration - previewAtTime );
+	const loopDurationHelp = createInterpolateElement(
+		sprintf(
+			/* translators: placeholder is the maxium lapse duration for the previewOnHover */
```
Comment (dhasilva) [inline projects/packages/videopress/src/client/block-editor/blocks/video/components/poster-panel/index.tsx:329]:
```suggestion
			/* translators: placeholder is the maximum lapse duration for the previewOnHover */
```

**Comment by retrofox** [comment]
closed in favor of https://github.com/Automattic/jetpack/pull/29850

**Pull request closed without merging**